zallet/src/commands/migrate_zcashd_wallet.rs

```diff
@@ -192,11 +193,59 @@ impl MigrateZcashdWalletCmd {
         chain_view: ChainView,
         wallet: ZcashdWallet,
         buffer_wallet_transactions: bool,
+        allow_multiple_wallet_imports: bool,
     ) -> Result<(), MigrateError> {
         let mut db_data = db.handle().await?;
         let network_params = *db_data.params();
         Self::check_network(wallet.network(), network_params.network_type())?;
 
+        let existing_zcash_sourced_accounts =
+            db_data
+                .get_account_ids()?
+                .into_iter()
+                .try_fold(vec![], |mut found, account_id| {
+                    let account = db_data
+                        .get_account(account_id)?
+                        .expect("account exists for just-retrieved id");
+
+                    match account.source() {
+                        AccountSource::Derived {
+                            derivation,
+                            key_source,
+                        } if key_source.as_ref() == Some(&ZCASHD_MNEMONIC_SOURCE.to_string()) => {
+                            found.push(derivation.clone());
+                        }
+                        _ => {}
+                    }
+
+                    Ok::<_, SqliteClientError>(found)
+                })?;
```

Comment:
`try_fold` is an odd way to do this. Why not:
```suggestion
                .filter_map(|account_id| {
                    match db_data
                        .get_account(account_id) {
                        Err(e) => Some(Err(e)),
                        Ok(account) => match account.expect("account exists for just-retrieved id")
                            .source() {
                            AccountSource::Derived {
                                derivation,
                                key_source,
                            } if key_source.as_ref() == Some(&ZCASHD_MNEMONIC_SOURCE.to_string()) => {
                                Some(Ok(derivation.clone()))
                            }
                            _ => None
                        },
                    }
                })
                .collect::<Result<Vec<_>, _>>()?;
```

Reply:
I don't see that this is particularly any clearer? `try_fold` is a very common way of building lists.

Reply:
I disagree for the original list, but after the switch to just collecting fingerprints into a `HashSet` (which has a different cardinality than the original sequence), `try_fold` feels fine.